docs/glossary.rst

```diff
@@ -0,0 +1,29 @@
+Glossary
+========
+
+.. glossary::
+
+    reservoir simulator
+        Reservoir simulators such as OPM Flow or Eclipse. 
```

Comment:
Kind of defining a word with the word itself here. Here is a suggestion:
```
Simulation of reservoir fields come in many forms, but for the purposes of res2df we 
only consider those who takes a :term:`.DATA` file as input and produces term`output file`s such
`.UNSRMY`.
```
Could probably also link to https://petrowiki.spe.org/Reservoir_simulation.

Reply:
Reworded, but made the input be specified to `deck`, as the simulators can accept a .DATA file, but also individual include files, so the scoping to .DATA only is too specific.